core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1150,7 +1150,6 @@ class GroupCoordinator(val brokerId: Int,
       group.notYetRejoinedMembers.filterNot(_.isStaticMember) foreach { failedMember =>
         removeHeartbeatForLeavingMember(group, failedMember)
         group.remove(failedMember.memberId)
-        group.removeStaticMember(failedMember.groupInstanceId)
```

Comment:
This is a cleanup. In the current logic we should never hit a static member in this block